### PR TITLE
Add Feature mappings

### PR DESCRIPTION
**Pull request by jaskarth**
Map the new features and fix some unmapped names in existing classes.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/class_4779 net/minecraft/world/gen/feature/BasaltPillarFeature
+	METHOD method_24434 setBasaltRandomly (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
+		ARG 1 world
+		ARG 3 pos
+	METHOD method_24435 didSetBasalt (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)Z
```
Comment (liach) [inline mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping]: Is this like "try set basalt" or `is basalt set`

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/structure/NetherFossilGenerator.mapping [hunk 2]
```diff
@@ -0,0 +1,19 @@
+CLASS net/minecraft/class_4787 net/minecraft/structure/NetherFossilGenerator
+	FIELD field_22197 FOSSILS [Lnet/minecraft/class_2960;
+	METHOD method_24453 addPieces (Lnet/minecraft/class_3485;Ljava/util/List;Ljava/util/Random;Lnet/minecraft/class_2338;)V
+		ARG 0 manager
+		ARG 1 pieces
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/structure/NetherFossilGenerator.mapping:7]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/structure/StructurePiece.mapping [hunk 3]
```diff
@@ -65,6 +65,9 @@ CLASS net/minecraft/class_3443 net/minecraft/structure/StructurePiece
 		ARG 8 lootTbaleId
 	METHOD method_14931 generate (Lnet/minecraft/class_1936;Lnet/minecraft/class_2794;Ljava/util/Random;Lnet/minecraft/class_3341;Lnet/minecraft/class_1923;)Z
 		ARG 1 world
+		ARG 2 generator
+		ARG 4 box
```
Comment (Juuxel) [inline mappings/net/minecraft/structure/StructurePiece.mapping:70]:
```suggestion
		ARG 3 random
		ARG 4 box
```

File: mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/class_4779 net/minecraft/world/gen/feature/BasaltPillarFeature
+	METHOD method_24434 tryPlaceBasalt (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
+		ARG 1 world
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping:5]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+	METHOD method_24435 stopOrPlaceBasalt (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)Z
+		ARG 1 world
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/BasaltPillarFeature.mapping:9]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/NoSurfaceOreFeature.mapping [hunk 6]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/class_4783 net/minecraft/world/gen/feature/NoSurfaceOreFeature
+	METHOD method_24446 checkAir (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;)Z
+		ARG 1 world
+		ARG 2 pos
+	METHOD method_24447 getStartPos (Lnet/minecraft/class_2338$class_2339;Ljava/util/Random;Lnet/minecraft/class_2338;I)V
+		ARG 1 mutable
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/NoSurfaceOreFeature.mapping:8]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/NoSurfaceOreFeature.mapping [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+		ARG 4 size
+	METHOD method_24448 randomCoord (Ljava/util/Random;I)I
+		ARG 2 size
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/NoSurfaceOreFeature.mapping:12]:
```suggestion
		ARG 1 random
		ARG 2 size
```

File: mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping [hunk 8]
```diff
@@ -0,0 +1,14 @@
+CLASS net/minecraft/class_4784 net/minecraft/world/gen/feature/WeepingVinesFeature
+	FIELD field_22196 DIRECTIONS [Lnet/minecraft/class_2350;
+	METHOD method_24450 generateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;III)V
+		ARG 0 world
+		ARG 2 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping:6]:
```suggestion
		ARG 1 random
		ARG 2 pos
```

File: mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+		ARG 3 length
+		ARG 4 minAge
+		ARG 5 maxAge
+	METHOD method_24451 tryGenerateNearVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
+		ARG 1 world
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping:13]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+	METHOD method_24452 tryGenerateFarVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
+		ARG 1 world
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/WeepingVinesFeature.mapping:17]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping [hunk 11]
```diff
@@ -0,0 +1,30 @@
+CLASS net/minecraft/class_4781 net/minecraft/world/gen/feature/HugeFungiFeature
+	METHOD method_24438 getStartPos (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2338$class_2339;
+		ARG 0 world
+		ARG 1 pos
+	METHOD method_24439 generateHatBlock (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_4780;Lnet/minecraft/class_2338$class_2339;FFF)V
+		ARG 1 world
+		ARG 3 config
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping:8]:
```suggestion
		ARG 2 random
		ARG 3 config
```

File: mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+		ARG 4 pos
+		ARG 5 stemHeight
+		ARG 6 thickStem
+	METHOD method_24441 tryGenerateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Z)V
+		ARG 1 world
+		ARG 3 pos
+		ARG 4 state
+	METHOD method_24442 generateVines (Lnet/minecraft/class_2338;Lnet/minecraft/class_1936;Ljava/util/Random;)V
+		ARG 1 pos
+		ARG 2 world
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping:27]:
```suggestion
		ARG 2 world
		ARG 3 random
```

File: mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+		ARG 7 vineChance
+	METHOD method_24440 generateStem (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_4780;Lnet/minecraft/class_2338$class_2339;IZ)V
+		ARG 1 world
+		ARG 3 config
+		ARG 4 pos
+		ARG 5 stemHeight
+		ARG 6 thickStem
+	METHOD method_24441 tryGenerateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Z)V
+		ARG 1 world
+		ARG 3 pos
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping:23]:
```suggestion
		ARG 2 random
		ARG 3 pos
```

File: mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+	METHOD method_24439 generateHatBlock (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_4780;Lnet/minecraft/class_2338$class_2339;FFF)V
+		ARG 1 world
+		ARG 3 config
+		ARG 4 pos
+		ARG 5 decorationChance
+		ARG 6 generationChance
+		ARG 7 vineChance
+	METHOD method_24440 generateStem (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_4780;Lnet/minecraft/class_2338$class_2339;IZ)V
+		ARG 1 world
+		ARG 3 config
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping:16]:
```suggestion
		ARG 2 random
		ARG 3 config
```

File: mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+	METHOD method_24441 tryGenerateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Z)V
+		ARG 1 world
+		ARG 3 pos
+		ARG 4 state
+	METHOD method_24442 generateVines (Lnet/minecraft/class_2338;Lnet/minecraft/class_1936;Ljava/util/Random;)V
+		ARG 1 pos
+		ARG 2 world
+	METHOD method_24443 generateHat (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_4780;Lnet/minecraft/class_2338$class_2339;IZ)V
+		ARG 1 world
+		ARG 3 config
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/feature/HugeFungiFeature.mapping:32]:
```suggestion
		ARG 2 random
		ARG 3 config
```

**Review by Juuxel — APPROVED** [review]
The PR looks good (though if you want to, you can add the `random` parameters with these suggestions).

**Review by Prospector — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**